packages/cache/src/store.js

```diff
@@ -77,10 +77,9 @@ const applyCacheTransaction = async (
 
   // Loop until our updated state is fresh wrt our current state.
   basisState = stateStore.get(keyStr);
-  const untilUpdateSynced = asyncGenerate(() => ({
-    value: null,
-    done: !updatedState || updatedState.generation > basisState.generation,
-  }));
+  const untilUpdateSynced = untilTrue(
+    () => !updatedState || updatedState.generation > basisState.generation,
+  );
```

Comment:
Please review carefully to see if I preserved equivalence. Something feels off.

Reply:
It's correct.  Stop when `updatedState === undefined` or when `updatedState.generation > basisState.generation`.